x-pack/legacy/plugins/apm/public/components/app/ServiceDetails/ServiceDetailTabs.tsx

```diff
@@ -10,60 +10,56 @@ import { IUrlParams } from '../../../context/UrlParamsContext/types';
 import { HistoryTabs } from '../../shared/HistoryTabs';
 import { ErrorGroupOverview } from '../ErrorGroupOverview';
 import { TransactionOverview } from '../TransactionOverview';
-import { ServiceMetrics } from './ServiceMetrics';
+import { ServiceMetrics } from '../ServiceMetrics';
+import { useFetcher } from '../../../hooks/useFetcher';
+import { loadServiceAgentName } from '../../../services/rest/apm/services';
+import { isRumAgentName } from '../../../../common/agent_name';
 
 interface Props {
-  transactionTypes: string[];
   urlParams: IUrlParams;
-  isRumAgent?: boolean;
-  agentName?: string;
 }
 
-export function ServiceDetailTabs({
-  transactionTypes,
-  urlParams,
-  isRumAgent,
-  agentName
-}: Props) {
-  const { serviceName } = urlParams;
-  const headTransactionType = transactionTypes[0];
+export function ServiceDetailTabs({ urlParams }: Props) {
```

Comment:
I prefer props because they are slightly easier to test and reason about. But OTOH it doesn't make sense to pass `urlParams` through many intermediary components that don't need it only to feed it to a leaf node. In those cases it's a no-brainer to change it to a hook.
For consistency one could argue that we should then always use hooks and never props. I don't think it has to be this black/white but not totally against it either.

So basically I'm fine with changing it as we see fit.